Assignmnet-4 _Sonar_Image_Analyzer.c for image processing

kirti734 opened the pull request.

kirti734 committed:
Add Sonar_Image_Analyzer.c for image processing

priyankagupta-ITT commented on Sonar_Image_Analyzer.c:
Enter matrix size (2-10): 3
Original
99 85 47
70 0 4
212 128 21

Rotated:
212 70 99
128 0 85
21 4 47

Final Output
102 99 63
72 74 50
38 47 34

Expected Output is 
103   99   64
 73   74   51
 38   48   34

Looks like output differs slightly due to rounding. Ensure rounding method aligns with expected results.

kirti734 replied:
I updated the code, use round() function to rounding the value so that it aligns with expected results.

kirti734 committed:
Use rounding for pixel value calculations

priyankagupta-ITT commented on Sonar_Image_Analyzer.c:
```diff
@@ -0,0 +1,200 @@
+		return 1;
+	}
+	
+	int originalMatrix[sizeofMatrix][sizeofMatrix];
```
You can't use array

kirti734 replied:
Now, I used the double pointers to store the matrix

kirti734 committed:
Refactor matrix functions for dynamic memory usage

Updated matrix functions to use double pointers for dynamic memory allocation and improved flexibility.

priyankagupta-ITT commented on Sonar_Image_Analyzer.c:
```diff
@@ -0,0 +1,217 @@
+	{
+		printf("Size OF Matrix should be in between 2 and 10");
+		return 1;
+	}
```
if (sizeofMatrix < 0)
	{
		printf("Size OF Matrix should be a positive value");
		return 1;
	}
	
	if(sizeofMatrix < MIN_MATRIX_SIZE || sizeofMatrix > MAX_MATRIX_SIZE)
	{
		printf("Size OF Matrix should be in between 2 and 10");
		return 1;
	}


do you think, we can merge both condition???

kirti734 replied:
Yes, I updated by combining these 2 condition

priyankagupta-ITT commented on Sonar_Image_Analyzer.c:
```diff
@@ -0,0 +1,217 @@
+	printf("Rotated:\n");
+	printMatrix(sizeofMatrix , originalMatrix);
+	
+	smoothingFilter(sizeofMatrix , originalMatrix);
```
rename this method like applySmoothingFilter()

kirti734 replied:
Now, I updated the function name from smoothingFilter() to applySmoothingFilter()

priyankagupta-ITT commented on Sonar_Image_Analyzer.c:
```diff
@@ -0,0 +1,217 @@
+	
+	int index;
+	
+	int **originalMatrix = malloc(sizeofMatrix * sizeof(int *));
```
How did you arrive at the decision to use a double pointer?

kirti734 replied:
I used doble pointer due to few reasons :-
-- It provides flexibility to handle variable-sized matrices easily.
-- It offers better readability and easier indexing, compare to single pointer.
-- It’s a standard and widely accepted way to handle dynamic 2D arrays in C.

kirti734 committed:
Update Sonar_Image_Analyzer.c
